modules/core/keeper/keeper.go

```diff
@@ -28,7 +28,7 @@ type Keeper struct {
 
 	cdc codec.BinaryCodec
 
-	ClientKeeper     clientkeeper.Keeper
+	ClientKeeper     *clientkeeper.Keeper
 	ConnectionKeeper connectionkeeper.Keeper
 	ChannelKeeper    channelkeeper.Keeper
 	PortKeeper       *portkeeper.Keeper
```

Comment:
@chatton and I have been discussing the mismatch of pointer and value types used here for a while, I think others are also aware of it - this was an issue with the `PortKeeper` below and caused problems with the router in the past.

@chatton volunteered to open an issue about this for discussion. I would be in favour of having consistency here, which would minimise the amount of debugging when things aren't quite lining up with what we expect to happen.

I think its probably best to not be dereferencing keepers when passing args all over the place - i.e. `*k.ClientKeeper` and such

Reply:
#6058